src/outpack/location.py

```diff
@@ -0,0 +1,108 @@
+
+LOCATION_LOCAL = "local"
+LOCATION_ORPHAN = "orphan"
+LOCATION_RESERVED_NAME = [LOCATION_LOCAL, LOCATION_ORPHAN]
+LOCATION_TYPES = [LOCATION_LOCAL, LOCATION_ORPHAN, "path"]
+
+
+def orderly_location_list(root=None, locate=True):
+    root = root_open(root, locate)
+    return list(root.config.location.keys())
+
+
+def orderly_location_add(name, type, args, root=None, locate=True):
+    root = root_open(root, locate)
+    # validate name is scalar character??
+    if name in LOCATION_RESERVED_NAME:
+        msg = f"Cannot add a location with reserved name '{name}'"
+        raise Exception(msg)
+
+    location_check_new_name(root, name)
+
+    loc = new_location_entry(name, type, args)
+
+    if type == "path":
+        ## We won't be necessarily be able to do this _generally_ but
+        ## here, let's confirm that we can read from the outpack archive
+        ## at the requested path; this will just fail but without
+        ## providing the user with anything actionable yet.
+        root_open(loc.args["path"], locate=False)
+
+    config = root.config
+    config.location[name] = loc
+    update_config(config, root.path)
+
+
+def orderly_location_remove(name, root=None, locate=True):
+    root = root_open(root, locate)
+
+    if name in LOCATION_RESERVED_NAME:
+        msg = f"Cannot remove default location '{name}'"
+        raise Exception(msg)
+
+    location_check_exists(root, name)
+    config = root.config
+
+    # TODO: mark packets as orphaned
+
+    location_path = root.path / ".outpack" / "location" / name
+    if location_path.exists():
+        ## Skipped on covr because this dir won't exist until packet pulling implemented
+        shutil.rmtree(location_path)  # pragma: no cover
+
+    # TODO: Rebuild the index after index has been added
+    config.location.pop(name)
+    update_config(config, root.path)
+
+
+def orderly_location_rename(old, new, root=None, locate=True):
+    root = root_open(root, locate)
+
+    if old in LOCATION_RESERVED_NAME:
+        msg = f"Cannot rename default location '{old}'"
+        raise Exception(msg)
+
+    location_check_new_name(root, new)
+    location_check_exists(root, old)
+
+    config = root.config
+    new_loc = config.location.pop(old)
+    new_loc.name = new
+    config.location[new] = new_loc
+    update_config(config, root.path)
+
+
+def location_check_new_name(root, name):
+    if location_exists(root, name):
+        msg = f"A location with name '{name}' already exists"
+        raise Exception(msg)
+
+
+def location_check_exists(root, name):
+    if not location_exists(root, name):
+        msg = f"No location with name '{name}' exists"
+        raise Exception(msg)
+
+
+def location_exists(root, name):
+    return name in orderly_location_list(root)
+
+
+def new_location_entry(name, type, args):
+    match_value(type, LOCATION_TYPES, "type")
+    if type == "path":
+        required = ["path"]
+    missing = set(required).difference(set(args.keys()))
+    if len(missing) > 0:
```

Comment:
```suggestion
    if missing:
```

pythonic way is to test for truthyness, with the empty set being falsy